### PR TITLE
Show battery warning in a snack bar

### PR DESCRIPTION
**Pull request by d-loose**
See https://github.com/canonical/ubuntu-desktop-installer/pull/1445#issuecomment-1441744399

### CODE REVIEW

**Comment by elioqoshi** [comment]
@d-loose Would appreciate if for every PR we can also have a screenshot for light and dark versions :)

File: packages/ubuntu_desktop_installer/lib/pages/updates_other_software/updates_other_software_page.dart [hunk 1]
```diff
@@ -100,10 +100,23 @@ class _UpdatesOtherSoftwarePageState extends State<UpdatesOtherSoftwarePage> {
           ),
         ],
       ),
-      footer: model.onBattery
-          ? Html(
-              data: lang.onBatteryWarning(
-                  Theme.of(context).colorScheme.error.toHex()),
+      snackBar: model.onBattery
+          ? SnackBar(
+              width: 500,
```
Comment (jpnurmi) [inline packages/ubuntu_desktop_installer/lib/pages/updates_other_software/updates_other_software_page.dart:105]: did you try what happens if the text is too long to fit? does it wrap?

**Comment by d-loose** [comment]
Screenshots:
| Light | Dark |
| - | - |
| image | image |

File: packages/ubuntu_desktop_installer/lib/pages/updates_other_software/updates_other_software_page.dart (hunk 1 above)
Comment (d-loose) [reply packages/ubuntu_desktop_installer/lib/pages/updates_other_software/updates_other_software_page.dart:105]: Yes, it wraps
image

**Review by jpnurmi — APPROVED** [review]
Sorry for the delay, LGTM! :+1: And thanks for splitting it off separately :)

**Comment by jpnurmi** [comment]
@elioqoshi Do you think we could replace the red "Warning" label with an icon? It could potentially look a bit nicer

**Comment by elioqoshi** [comment]
I think using an icon is fine here as well.

**Comment by d-loose** [comment]
> The theming is off I think. Ideally they should be reverted with the snackbar background color slightly different from the screen color. What do you think?

It's probably an issue for yaru. In principle I agree with you, although this is currently in accordance with the [M3 specs](https://m3.material.io/components/snackbar/specs#3c1ed6ed-2314-408b-aa75-634de186ec0a).

**Comment by jpnurmi** [comment]
@elioqoshi Do you think we could move forward with this or would you prefer we try to tweak the colors in yaru.dart first?

**Comment by elioqoshi** [comment]
I think we should solve the root issue and not the overlying issue here so we should go for this as it is now and address yaru.dart separately, what do you think?

**Comment by jpnurmi** [comment]
Ok, thanks! I think it looks decent enough to be on the daily image like this for a while. It's hard to say how long it might take to address this in yaru.dart. :)

**Comment by elioqoshi** [comment]
@jpnurmi do you think we can address it in yaru.dart before 23.04 release?

**Comment by jpnurmi** [comment]
@elioqoshi we can propose changes right away. would the looks of GNOME notifications work for us?

image

**Comment by elioqoshi** [comment]
I think that would work yeah. I can finetune the design if needed.

**Pull request merged**